### PR TITLE
[Log4Net]: Adds snippet showing how to set Log4Net properties as Google Cloud log entry labels.

### PR DESCRIPTION
**Pull request by amanda-tarafa**
Towards #5325

### CODE REVIEW

File: apis/Google.Cloud.Logging.Log4Net/Google.Cloud.Logging.Log4Net.Snippets/GoogleStackdriverAppenderSnippets.cs [hunk 1]
```diff
@@ -110,7 +110,86 @@ public void Overview()
...
+            xml = xml.Replace("PROJECT_ID", projectId).Replace("LOG_ID", logId);
+            Assert.False(File.Exists(fileName), "Test would overwrite existing file");
+            File.WriteAllText(fileName, xml);
+
+            try
+            {
+                // Resource: log4net-custom-labels-template.xml log4net_custom_labels_template
+                // Sample: Custom_Labels
+                // Configure log4net to use Google Cloud Logging from the XML configuration file.
+                XmlConfigurator.Configure(LogManager.GetRepository(GetType().GetTypeInfo().Assembly), new FileInfo("log4net.xml"));
```
Comment (jskeet) [inline apis/Google.Cloud.Logging.Log4Net/Google.Cloud.Logging.Log4Net.Snippets/GoogleStackdriverAppenderSnippets.cs:150]: I suspect `GetTypeInfo()` can be removed everywhere now - we only didn't have it in netstandard1.*

File: apis/Google.Cloud.Logging.Log4Net/Google.Cloud.Logging.Log4Net.Snippets/GoogleStackdriverAppenderSnippets.cs [hunk 2]
```diff
@@ -110,7 +110,86 @@ public void Overview()
...
+
+                // Retrieve a logger for this context.
+                ILog log = LogManager.GetLogger(typeof(Program));
+                // Set a property using one of the GlobaContext, ThreadContext or LogicalThreadContext classes.
+                LogicalThreadContext.Properties["property_for_label"] = "my label value";
+                // Log some information. This log entry will be sent to Google Cloud Logging with a label
+                // {"label_from_property", "my label value"}
```
Comment (jskeet) [inline apis/Google.Cloud.Logging.Log4Net/Google.Cloud.Logging.Log4Net.Snippets/GoogleStackdriverAppenderSnippets.cs:157]: Possibly explain that the mapping from "property_for_label" to "label_from_property" is part of the configuration? I thought this was a typo at first.

**Review by jskeet — APPROVED** [review]
A couple of suggestions, but nothing major.

File: apis/Google.Cloud.Logging.Log4Net/Google.Cloud.Logging.Log4Net.Snippets/GoogleStackdriverAppenderSnippets.cs (hunk 1 above)
Comment (amanda-tarafa) [reply apis/Google.Cloud.Logging.Log4Net/Google.Cloud.Logging.Log4Net.Snippets/GoogleStackdriverAppenderSnippets.cs]: Ah yes, will remove from all snippets.

File: apis/Google.Cloud.Logging.Log4Net/Google.Cloud.Logging.Log4Net.Snippets/GoogleStackdriverAppenderSnippets.cs (hunk 2 above)
Comment (amanda-tarafa) [reply apis/Google.Cloud.Logging.Log4Net/Google.Cloud.Logging.Log4Net.Snippets/GoogleStackdriverAppenderSnippets.cs]: Yep!

**Review by amanda-tarafa — COMMENTED** [review]
Comments adressed. Will merge on green save objections. Thanks!

**Pull request merged**